lib/common/zstd_internal.h

```diff
@@ -137,7 +137,10 @@ static const U32 OF_defaultNormLog = 5;
 /*-*******************************************
 *  Shared functions to include for inlining
 *********************************************/
-static void ZSTD_copy8(void* dst, const void* src) { memcpy(dst, src, 8); }
+static void ZSTD_copy8(void* dst, const void* src)
+{
+    MEM_write64(dst, MEM_read64(src));
```

Comment:
Something I did not took time to describe last year while redacting 
http://fastcompression.blogspot.fr/2015/08/accessing-unaligned-memory.html

is that `memcpy(dst, src, 8);` would always work correctly, on all architectures and compiler tested.

Problems were limited to `U64 val; memcpy(&val, src, 8);` which, in some cases, generated some completely useless store + retrieve intermediate operations. Hence the different tricks (`__packed`, direct `U64*` ) now present in `mem.h` to avoid these inefficiencies.

As an example :
https://godbolt.org/g/h1ODwa